src/DeterministicNoiseSimulator.cpp

```diff
@@ -470,36 +538,18 @@ std::map<std::string, double> DeterministicNoiseSimulator::AnalyseState(dd::Qubi
             }
             current_result = current_result >> 1;
         }
-        measure_result.insert({result_string, global_probability});
+        if (global_probability > 0.01) {
+            measuredResult.insert({result_string, global_probability});
+        }
     }
-    return measure_result;
+    return measuredResult;
 }
 
-void DeterministicNoiseSimulator::generateGate(qc::MatrixDD* pointer_for_matrices, char noise_type, dd::Qubit target) {
+void DeterministicNoiseSimulator::generateGate(qc::MatrixDD* pointer_for_matrices, char noise_type, dd::Qubit target, double probability) {
```

Comment:
This routine should probably reside in the QFR as do all the other routines that generate decision diagrams for gates.
Furthermore, use an appropriate type (such as `std::array` or `std::vector`) instead of the ugly pointer type that hides an array underneath.

Reply:
done. Changed to:
> std::array<mEdge, std::tuple_size_v<decltype(dd::dNode::e)>> &pointerForMatrices,